reframe/core/buildsystems.py

```diff
@@ -865,7 +865,7 @@ def emit_build_commands(self, environ):
         return ret
 
     def _env_activate_cmds(self):
-        cmds = ['. $SPACK_ROOT/share/spack/setup-env.sh']
+        cmds = ['. "$(spack location --spack-root)/share/spack/setup-env.sh"']
```

Comment:
Do we need the quotes here? I would go without unless you have a better reason.

Reply:
To protect the expansion of `$(spack location --spack-root)` in case it contains whitespaces.  I'm actually very surprised double quotes aren't used anywhere in ReFrame :slightly_smiling_face:

Reply:
For example, try to put
```sh
#!/bin/bash
. $(spack location --spack-root)/share/spack/setup-env.sh
```
in https://www.shellcheck.net/.  See also https://github.com/koalaman/shellcheck/wiki/SC2046

Reply:
Thanks for the inputs! I'll open an issue for shell checking all the generated scripts from ReFrame. 

> I'm actually very surprised double quotes aren't used anywhere in ReFrame 🙂

Generally, the concept has always been in ReFrame to generate as simple shell scripts as possible and have any more complicated logic in the framework. Actually, I think that this is the first time the framework emits a command substitution in the script :) Also, I wouldn't like to quote user commands that are meant to be inserted verbatim, such as  `prerun_cmds` etc.